### PR TITLE
API acceptance tests for WebDAV search

### PR DESCRIPTION
**Pull request by individual-it**
## Description
API tests to test webdav search see https://github.com/owncloud/core/pull/31873

## Related Issue
https://github.com/owncloud/search_elastic/issues/26

## How Has This Been Tested?
run the tests locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [x] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 1]
```diff
@@ -2455,4 +2458,68 @@ public function userRestoresVersionIndexOfFile($user, $versionIndex, $path) {
...
+	 * parses a PROPFIND response from $this->response into xml
+	 * and searches for a given filename in the response list
+	 *
+	 * @param string $user
+	 * @param string $fileNameToSearch
+	 *
+	 * @return string or false if file could not be found
+	 */
+	public function findFileFromPropfindResponse($user, $fileNameToSearch) {
+		//if we using that step the second time in a scenario e.g. 'But ... should not'
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebDav.php]: if we are using the step the second time...

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 2]
```diff
@@ -2455,4 +2458,68 @@ public function userRestoresVersionIndexOfFile($user, $versionIndex, $path) {
...
+		//then don't parse the result again, because the result in a ResponseInterface
+		//and can only be read once
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebDav.php]: delete "and"

**Review by phil-davis — APPROVED** [review]
Looks good.
That ``['value'][1]['value'][0]['value']`` stuff is ugly, but what  to do about that?

**Comment by individual-it** [comment]
backport in #32818

**Comment by phil-davis** [comment]
stupid thing, no codecov has come so we cannot merge.

**Comment by phil-davis** [comment]
Now it failed
```
tests/acceptance/features/webUISharingExternal/shareByPublicLink.feature:153
```
That scenario seems to fail sometimes. I guess there is some timing problem with the public link dialog.

**Pull request merged**